components/TicTacToe.jsx

```diff
@@ -7,22 +7,64 @@ export const TicTacToe = () => {
     border: "1px solid",
     boxSizing: "content-box",
   };
+  const cols = 3;
+  const rows = 3;
   const pieces = ["X", "O"];
   const [turn, setTurn] = useState(0);
   const [grid, setGrid] = useState(Array(9).fill(null));
+  const [win, setWin] = useState("in progress");
+
+  const notClicked = (i) => {
+    if (pieces.indexOf(grid[i]) < 0) {
+      return true;
+    }
+    return false;
+  };
+
+  const gridIndex = (x, y) => {
+    return cols * x + y;
+  };
+
+  const checkWinHorizontal = () => {
+    for (let x = 0; x < rows; x += 1) {
+      for (let y = 0; y < cols; y += 1) {
+        if (
+          grid[gridIndex(x, y)] == grid[gridIndex(x, y + 1)] &&
+          grid[gridIndex(x, y + 1)] == grid[gridIndex(x, y + 2)] &&
+          grid[gridIndex(x, y)] != null
```

Comment:
This form of check only works for a win condition of 3-in-a-row. If you wanted to change it to 2-in-a-row or N-in-a-row, how would you do this?

What if the win condition was `N = 10-in-a-row`?